---
title: 'HiCcompare: an R-package for joint normalization of Hi-C datasets and differential chromatin interaction detection'
csl: styles.ref/bioinformatics.csl
fontsize: 12pt
output:
  word_document:
    reference_docx: styles.doc/bioinformatics_styles_singlespace.docx
  pdf_document:
    fig_caption: no
  html_document: null
bibliography: /Users/mdozmorov/Documents/Work/VCU_work/3D_DNA/manuscript/3D_DNA-references.bib
---

John C. Stansfield^1,3^, Kellen G. Cresswell^1^, Vladimir I. Vladimirov^2^, Mikhail G. Dozmorov^1,3\*^ 

^1^ Dept. of Biostatistics, Virginia Commonwealth University, Richmond, VA, 23298, USA  
^2^ Dept. of Psychiatry, Virginia Institute for Psychiatric and Behavioral Genetics, Richmond, VA, 23219, USA  
^3^ Equal contribution  
^\*^ To whom correspondence should be addressed


# Abstract

**Motivation:** Changes in spatial chromatin interactions are now emerging as a unifying mechanism orchestrating the regulation of gene expression. Hi-C sequencing technology now allows insight into chromatin interactions on a genome-wide scale. However, Hi-C data contains many DNA sequence- and technology-driven biases. These biases prevent effective comparison of chromatin interactions aimed at identifying genomic regions differentially interacting between, disease-normal states or different cell types. Several methods have been developed for normalizing individual Hi-C datasets. However, they fail to account for biases _between two or more Hi-C datasets_, hindering comparative analysis of chromatin interactions.   
**Results:** We developed a simple and effective method, `HiCcompare`, for the joint normalization and differential analysis of multiple Hi-C datasets.  The method avoids constraining Hi-C data within a rigid statistical model, allowing a data-driven normalization of biases using locally weighted linear regression (`loess`). The method identifies region-specific chromatin interaction changes complementary to changes due to large-scale genomic rearrangements, such as copy number variants (CNVs). `HiCcompare` outperforms methods for normalizing individual Hi-C datasets in detecting _a priori_ known chromatin interaction differences in simulated and real-life settings while preserving the detection of genomic structures, such as A/B compartments.   
**Availability:** `HiCcompare` is freely available as a Bioconductor R package [https://bioconductor.org/packages/HiCcompare/](https://bioconductor.org/packages/HiCcompare/).  
**Contact:** mikhail.dozmorov@vcuhealth.org  
**Supplementary information:** Supplementary data are available at _Bioinformatics_ online.