---
output:
  pdf_document:
    toc: no
csl: /Users/mdozmorov/Documents/Work/VCU_work/3D_DNA/HiCcompare/styles.ref/bioinformatics.csl
bibliography: /Users/mdozmorov/Documents/Work/VCU_work/3D_DNA/manuscript/3D_DNA-references.bib
---

\pagenumbering{gobble}

# 2 Distance-centric concept of chromatin interaction frequencies

**Figure. Chromatin interaction frequency representation and dependence on distance.** (A) Distance-centric (off-diagonal) view of chromatin interaction matrices. Values on the diagonal represent interaction frequencies of regions at distance 0 (self-interacting). Each off-diagonal vector of interaction frequencies represents interactions at a given distance between pairs of regions. Triangles mark pairs of genomic regions interacting at the same distance. Data for chromosome 1, K562 cell line, 50KB resolution, spanning 0 - 7.5Mb is shown. (B) Deviation from the ideal power-law relationship (straight lines) between the $log_{10}-log_{10}$ interaction frequencies and distance, observed by others [@Lieberman-Aiden:2009aa; @Sanborn:2015aa; @Tanizawa:2010aa; @Jin:2013aa; @Cairns:2016aa; @Lun:2015aa; @Di-Stefano:2016aa; @Sexton:2012aa]. Curved lines represent chromosome-specific `loess` fits of the relationship, colored by datasets. The full range of genomic distances is shown. Data from HMEC, IMR90, NHEK cell lines, using all chromosomes, 500kb resolution were used.

\begin{center}
\includegraphics[height=250px]{../figures/fig1.png}
\end{center}

